source/common/quic/platform/BUILD

```diff
@@ -122,28 +121,38 @@ envoy_cc_library(
 )
 
 envoy_cc_library(
-    name = "quic_platform_impl_lib",
+    name = "quic_platform_hostname_utils_impl_lib",
     srcs = [
         "quic_hostname_utils_impl.cc",
     ],
     hdrs = [
         "quic_hostname_utils_impl.h",
-        "quic_mutex_impl.h",
     ],
     external_deps = [
-        "quiche_quic_platform_base",
         "abseil_str_format",
-        "abseil_synchronization",
-        "abseil_time",
-        "ssl",
+        "quiche_quic_platform_base",
     ],
     tags = ["nofips"],
     visibility = ["//visibility:public"],
     deps = [
-        "//source/common/common:assert_lib",
-        "//source/common/filesystem:directory_lib",
-        "//source/common/filesystem:filesystem_lib",
         "//source/common/http:utility_lib",
+        "@com_github_google_quiche//:quic_platform_export",
+    ],
+)
+
+envoy_cc_library(
+    name = "quic_platform_mutex_impl_lib",
```

Comment:
Consider renaming this target to :quiche_platform_mutex_impl_lib.

Reply:
I've removed this entire target now that we're using the common impl (as per your great observation!).